### ISSUE TITLE
Should data uri be allowed as logo_uri ?

### ISSUE THREAD

**Issue by matthieusieben** [aaronpk/draft-parecki-oauth-client-id-metadata-document#19]
If so, how would it impact the size limit of the whole metadata document ?

**Comment by ThisIsMissEm** [comment]
I would argue that data URIs should *not* be used in the metadata documents, and only `https` URIs should be used. If you did wish to support data URIs, then it would obviously increase the size limit you'd need for the whole metadata document .

That's why I went with a non-binding size limit recommendation of 5kb, because I'm aware that there can be certain scenarios where the document may be bigger, particularly with localisation of fields and such.

**Comment by matthieusieben** [comment]
Well, one advantage of data-uri is that it is easier to let the AS "allow list" a particular image for a particular client, and show that image to the user while logging in.

When an HTTPS uri is used, the actual file served could be different for the AS and USER, which could help performing phishing attacks.

**Comment by matthieusieben** [comment]
But I agree that the effect on the document size is not ideal...

**Comment by ThisIsMissEm** [comment]
> When an HTTPS uri is used, the actual file served could be different for the AS and USER, which could help performing phishing attacks.

This is true, but linking directly to a remote file on a security sensitive page would not be advisable; Instead I'd recommend AS's cache media assets such as that

**Comment by ThisIsMissEm** [comment]
Have opened #25

**Comment by ThisIsMissEm** [comment]
I don't specifically call out data uri's in that security consideration, but more that the authorization server shouldn't just be using the `logo_uri` verbatim in their authorization pages, and instead should fetch the resource and cache it, as to serve it from a trusted domain (this would also be inline with having a narrow CSP for `image-src`)

**Comment by aaronpk** [comment]
I agree with the recommendation for the AS to cache and serve the image itself. 

Interestingly, the client providing a data URI and the AS outputting the string verbatim does meet the requirement of not serving 3rd party hosted images 😂
